### DISCUSSION TITLE
Feature Request: RPR ML Denoiser and Viewport Upscaling available even when not using RPR

### DISCUSSION THREAD

**Discussion by OneDayCreations** [GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon#655] [Ideas]
Might seem a little odd, but I would love it if RPR's ML Denoiser and Viewport Upscaling were available with other render engines. Like any render engine RPR isn't always the best tool for the job, however it has some amazing features that could benefit all projects. 
While Blender natively has Intel's OIDN implemented it would be great to have an alternative, particularly one designed for AMD GPUs. 
And the viewport upscaling feature speaks for itself, interactive render mode is spectacular, but same principle, it would be good to have access to this tool without having to commit to a particular render engine. 

Thank you for all your work thus far =)

**Comment by ARozghon** [comment]
hi, that's weird because it was deactivated, maybe, this fix is not available yet in the release builds 
![image](https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon/assets/68949117/63067f5e-d1ac-4c7c-8fc9-42806fcc4102)

**Reply by OneDayCreations** [reply]
Hey, thanks for the reply, I'm away at the moment but I can have a look at this when I am home and let you know blender and rpr versions ect. 

But was hoping that the denoiser and upscaling features could be exposed to use with cycles or eevee -insert prefered renderer here-

**Comment by bsavery** [comment]
> Might seem a little odd, but I would love it if RPR's ML Denoiser and Viewport Upscaling were available with other render engines. Like any render engine RPR isn't always the best tool for the job, however it has some amazing features that could benefit all projects. While Blender natively has Intel's OIDN implemented it would be great to have an alternative, particularly one designed for AMD GPUs. And the viewport upscaling feature speaks for itself, interactive render mode is spectacular, but same principle, it would be good to have access to this tool without having to commit to a particular render engine.
> 
> Thank you for all your work thus far =)

Hello.  This is not something we can do in a Python addon like this.  However I would note that there is some effort in blender's side to add GPU denoising support.

**Reply by OneDayCreations** [reply]
No worries:) thanks you anyway, I don't know what limitations exist around addon vs native features, or between various scripting languages so I am always happy to learn something new. 

Thank you again for your work on RPR, always looking forward to how it develops,
All the best :)
